### PR TITLE
JAMES-1981 Don't fail when indexing fails

### PR DESCRIPTION
**Pull request by aduprat**

### CODE REVIEW

File: backends-common/elasticsearch/src/main/java/org/apache/james/backends/es/ElasticSearchIndexer.java [hunk 1]
```diff
@@ -103,26 +105,34 @@ public IndexResponse indexMessage(String id, String content) {
...
+            Preconditions.checkNotNull(updatedDocumentParts);
+            BulkRequestBuilder bulkRequestBuilder = client.prepareBulk();
+            updatedDocumentParts.forEach(updatedDocumentPart -> bulkRequestBuilder.add(
+                client.prepareUpdate(
+                    indexName.getValue(),
+                    typeName.getValue(),
+                    updatedDocumentPart.getId())
+                .setDoc(updatedDocumentPart.getUpdatedDocumentPart())));
+            return Optional.of(bulkRequestBuilder.get());
+        } catch (ValidationException e) {
```
Comment (rouazana) [inline backends-common/elasticsearch/src/main/java/org/apache/james/backends/es/ElasticSearchIndexer.java]: No logging?
Do you know why the indexing can fail?
If the exception is issued during the forEach, it means other parts are ignored?

Comment (chibenwa) [inline backends-common/elasticsearch/src/main/java/org/apache/james/backends/es/ElasticSearchIndexer.java]: Quynh pointed out the fact it was due to Empty list updates.

Could we have a shortcut if to handle this special case?

Comment (aduprat) [inline backends-common/elasticsearch/src/main/java/org/apache/james/backends/es/ElasticSearchIndexer.java]: @rouazana 
1/ Logging -> Ok
2/ fails in *BulkRequestBuilder.get* which executes the query

@chibenwa 
We have to be more generic, and never fails if indexing fails.

**Comment by aduprat** [comment]
> Could we also ensure not firing empty events? The right place for this is IMO MailboxEventDispatcher. We could add unit tests there too.

Ok

**Comment by chibenwa** [comment]
Just merged

**Pull request closed without merging**